src/main/java/com/musialowski/sqlcontainer/SqlFilesProcessor.java

```diff
@@ -29,33 +29,49 @@ public SqlFilesProcessor(Log logger) {
+            try (InputStream sqlFileInputStream = new FileInputStream(sqlFile)) {
+                fileLines = IOUtils.readLines(sqlFileInputStream);
+            }
+        } catch (IOException e) {
+            logger.error(e.getLocalizedMessage(), e);
             return false;
         }
-        StringBuffer sqlStatementBuffer = new StringBuffer();
+
+        Set<SqlStatementFieldVar> sqlStatementFieldVars = new HashSet<>();
+        StringBuffer sqlStatementCommentBuffer = new StringBuffer();
+        StringBuffer sqlStatementValueBuffer = new StringBuffer();
         String lastestKey = "";
         for (String line : fileLines) {
-            line = StringUtils.remove(line, ";");
-            if (line.startsWith(METHOD_NAME_PREFIX)) {
-                methodNameReturnedStringMap.put(lastestKey, sqlStatementBuffer.toString());
-                sqlStatementBuffer = new StringBuffer();
-                String methodName = StringUtils.removeStart(line, METHOD_NAME_PREFIX);
+            if (line.startsWith(DEFAULT_METHOD_DEFINITION_PREFIX)) {
+                if (StringUtils.isNotBlank(lastestKey)) {
+                    sqlStatementFieldVars.add(new SqlStatementFieldVar(lastestKey, removeLastSqlStatementSeparator(sqlStatementCommentBuffer.toString().trim()), inlineSqlStatement(removeLastSqlStatementSeparator(sqlStatementValueBuffer.toString().trim()))));
+                }
+                sqlStatementValueBuffer = new StringBuffer();
+                sqlStatementCommentBuffer = new StringBuffer();
+                String methodName = prepareSqlStatementName(line);
                 lastestKey = methodName;
-                methodNameReturnedStringMap.put(methodName, SQL_PLACEHOLDER);
                 continue;
             }
-            sqlStatementBuffer.append(line).append(NEW_LINE);
+            sqlStatementCommentBuffer.append(line).append(DEFAULT_EOF);
+            sqlStatementValueBuffer.append(line.trim()).append(" ");
         }
-        methodNameReturnedStringMap.put(lastestKey, sqlStatementBuffer.toString());
-        methodNameReturnedStringMap.remove("");
+        sqlStatementFieldVars.add(new SqlStatementFieldVar(lastestKey, removeLastSqlStatementSeparator(sqlStatementCommentBuffer.toString().trim()), inlineSqlStatement(removeLastSqlStatementSeparator(sqlStatementValueBuffer.toString().trim()))));
+
         SqlClassGenerator sqlClassGenerator = new SqlClassGenerator(logger);
-        sqlClassGenerator.generateSqlContainerClass(FilenameUtils.removeExtension(sqlFile.getName()), packageName, outputDirectory, methodNameReturnedStringMap);
+        sqlClassGenerator.generateSqlContainerClass(FilenameUtils.removeExtension(sqlFile.getName()), packageName, outputDirectory, sqlStatementFieldVars);
         return true;
     }
+
+    private String removeLastSqlStatementSeparator(String sqlStatement) {
+        return StringUtils.reverse(StringUtils.reverse(sqlStatement).replaceFirst(DEFAULT_SQL_STATEMENT_SEPARATOR, ""));

```

Comment:
Nie byłoby bezpieczniej i wydajniej użyć takiego kodu jak na dole?
```Java
if (sqlStatement.length() > 0 && sqlStatement.charAt(str.length()-1)==DEFAULT_SQL_STATEMENT_SEPARATOR) {
      sqlStatement = sqlStatement.substring(0, sqlStatement.length()-1);
}
```
Albo jeżeli dopuszczamy, że zakończenie zapytania może zawierać na końcu np. komentarz to użyć:
```Java
StringBuffer stringBuffer = new StringBuffer(sqlStatement);
stringBuffer.deleteCharAt(sqlStatement.lastIndexOf(";"));
```